inbm/dispatcher-agent/dispatcher/fota/bios_factory.py

```diff
@@ -208,15 +221,7 @@ def _apply_firmware(self, repo_name: str, fw_file: Optional[str], manifest_guid:
         """
         guid = ''
         if self._guid_required:
-            # get the GUID from the system using FW tool
-            extracted_guids = extract_guids(
-                self._fw_tool, ["System Firmware type", "system-firmware type"])
-            if manifest_guid:
-                if manifest_guid not in extracted_guids:
-                    raise FotaError(
-                        f"GUID in manifest does not match any system firmware GUID on the system")
-                else:
-                    guid = manifest_guid
+            guid = self._get_guid_from_system(manifest_guid)
 
         if not tool_options:
             tool_options = ''
```

Comment:
<h3>$\color{#FFA500}Improvements: Code Refactoring$</h3>

Remove the redundant empty string assignments for `tool_options` and `fw_file` by using default argument values in the method definition.